Jellyfin.Api/Controllers/DevicesController.cs

```diff
@@ -111,28 +114,34 @@ public async Task<ActionResult> UpdateDeviceOptions(
     }
 
     /// <summary>
-    /// Deletes a device.
+    /// Deletes devices.
     /// </summary>
-    /// <param name="id">Device Id.</param>
+    /// <param name="id">Optional. Device Id.</param>
+    /// <param name="ids">Device Ids.</param>
     /// <response code="204">Device deleted.</response>
     /// <response code="404">Device not found.</response>
-    /// <returns>A <see cref="NoContentResult"/> on success, or a <see cref="NotFoundResult"/> if the device could not be found.</returns>
+    /// <returns>A <see cref="NoContentResult"/> on success, or a <see cref="NotFoundResult"/> if a device could not be found.</returns>
     [HttpDelete]
     [ProducesResponseType(StatusCodes.Status204NoContent)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
-    public async Task<ActionResult> DeleteDevice([FromQuery, Required] string id)
+    public async Task<ActionResult> DeleteDevice([FromQuery, ParameterObsolete] string? id, [FromQuery, ModelBinder(typeof(CommaDelimitedArrayModelBinder))] string[] ids)
     {
-        var existingDevice = _deviceManager.GetDevice(id);
-        if (existingDevice is null)
+        string[] allIDs = ids.Concat(id != null ? [id] : Array.Empty<string>()).ToArray();
+
+        foreach (var i in allIDs)
         {
-            return NotFound();
-        }
+            var existingDevice = _deviceManager.GetDevice(i);
+            if (existingDevice is null)
+            {
+                return NotFound();
```

Comment:
This really is a no-go. You should first check all devices for existence before proceeding with the delete operation. Otherwise the client can never know which devices were deleted.